.github/workflows/validate-theme-entry.yml

```diff
@@ -0,0 +1,263 @@
+
+                if (owner.toLowerCase() !== author.toLowerCase()) {
+                    try {
+                        const isInOrg = await github.rest.orgs.checkMembershipForUser({ org: owner, username: author });
+                        if (!isInOrg) {
+                            throw undefined;
+                        }
+                    } catch (e) {
+                        addError(`The newly added entry is not at the end, or you are submitting on someone else's behalf. The last plugin in the list is: \`${plugin.repo}\`.`);
+                    }
+                }
+                try {
+                    const repository = await github.rest.repos.get({ owner, repo });
+                    if (!repository.data.has_issues) {
+                        addWarning('Your repository does not have issues enabled.');
+                    }
+                } catch (e) {
+                    addError(`It seems like you made a typo in the repository field ${theme.repo}`);
+                    return;
+                }
+
+                let manifest = {};
+
+                try {
+                    let manifestFile = await github.rest.repos.getContent({
+                        owner,
+                        repo,
+                        path: 'manifest.json',
+                    });
+
+                    manifest = JSON.parse(Buffer.from(manifestFile.data.content, 'base64').toString('utf-8'));
+                } catch (e) {
+                    addError(`You don't have a valid \`manifest.json\` at the root of your repo.`);
+                    return;
+                }
+
+                let validManifestKeys = ['name', 'minAppVersion', 'author', 'authorUrl'];
+                for (let key in manifest) {
+                    if (!validKeys.includes(key) && !key === 'version') {
+                        addError(`Your manifest has the invalid \`${key}\` property.`);
+                    }
+                }
+
+                for (let key in validManifestKeys) {
+                    if (!manifest.hasOwnProperty(key)) {
+                        addError(`Your manifest does not have the required \`${key}\` property.`);
+                    }
+                }
+
+                if (manifest.hasOwnProperty('name') && manifest.name != theme.name) {
+                    addError(`Plugin name mismatch, the name in this PR (\`${theme.name}\`) is not the same as the one in your repo (\`${manifest.name}\`). If you just changed your plugin name, remember to change it in the manifest.json in your repo and your latest GitHub release.`);
+                }
+
+                if (manifest.hasOwnProperty('version') && !(/^[0-9.]+$/i.test(manifest.version))) {
+                    addError('Your latest version number is not valid. Only numbers and dots are allowed.');
+                }
+
+                if (manifest.hasOwnProperty('author') && /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/.test(manifest.author)) {
```

Comment:
The author check should be moved earlier and should run on the submitted object rather than the manifest.